### PR TITLE
add swagger spec

### PR DESCRIPTION
**Pull request by darkonie**

### CODE REVIEW

**Comment by malnick** [comment]
Pending review from Joel...

File: swagger.yml [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+  skip_prev:
+    name: skip_prev
+    in: query
+    description: Move cursor position backward.
+    required: false
+    type: integer
+  cursor:
+    name: cursor
+    in: query
+    description: Move current cursor to provided cursor position
```
Comment (joel-hamill) [inline swagger.yml]: Add period

File: swagger.yml [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    required: false
+    type: string
+  read_reverse:
+    name: read_reverse
+    in: query
+    description: Read the journal entries in reverse. Default false
```
Comment (joel-hamill) [inline swagger.yml]: `The default value is false.`

File: swagger.yml [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+      parameters:
+        - $ref: "#/parameters/filter"
+        - $ref: "#/parameters/limit"
+        - $ref: "#/parameters/skip_next"
+        - $ref: "#/parameters/skip_prev"
+        - $ref: "#/parameters/cursor"
+        - $ref: "#/parameters/read_reverse"
+      responses:
+        200:
+          description: Successful response
```
Comment (joel-hamill) [inline swagger.yml]: Add period

File: swagger.yml [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+          schema:
+            title: logs Entries
+            type: string
+        201:
+          description: No logs found
+        401:
+          description: Not authorized
```
Comment (joel-hamill) [inline swagger.yml]: Add period

File: swagger.yml [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+        - $ref: "#/parameters/cursor"
+        - $ref: "#/parameters/read_reverse"
+      responses:
+        200:
+          description: Successful response
+          schema:
+            title: logs Entries
+            type: string
+        201:
+          description: No logs found
```
Comment (joel-hamill) [inline swagger.yml]: Add period

File: swagger.yml [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+        401:
+          description: Not authorized
+        500:
+          description: Internal server error
```
Comment (joel-hamill) [inline swagger.yml]: Add period

File: swagger.yml [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+        - $ref: "#/parameters/filter"
+        - $ref: "#/parameters/limit"
+        - $ref: "#/parameters/skip_next"
+        - $ref: "#/parameters/skip_prev"
+        - $ref: "#/parameters/cursor"
+        - $ref: "#/parameters/read_reverse"
+        - $ref: "#/parameters/postfix"
+      responses:
+        200:
+          description: Successful response
```
Comment (joel-hamill) [inline swagger.yml]: Add period

File: swagger.yml [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+          schema:
+            title: logs Entries
+            type: string
+        201:
+          description: No logs found
```
Comment (joel-hamill) [inline swagger.yml]: Add period

File: swagger.yml [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+        401:
+          description: Not authorized
```
Comment (joel-hamill) [inline swagger.yml]: Add period

File: swagger.yml [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+        500:
+          description: Internal server error
```
Comment (joel-hamill) [inline swagger.yml]: Add period

File: swagger.yml [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+      parameters:
+        - $ref: "#/parameters/filter"
+        - $ref: "#/parameters/limit"
+        - $ref: "#/parameters/skip_next"
+        - $ref: "#/parameters/skip_prev"
+        - $ref: "#/parameters/cursor"
+        - $ref: "#/parameters/read_reverse"
+      responses:
+        200:
+          description: Successful response
```
Comment (joel-hamill) [inline swagger.yml]: Add period

File: swagger.yml [hunk 12]
```diff
@@ -0,0 +1,218 @@
...
+          schema:
+            title: logs Entries
+            type: string
+        201:
+          description: No logs found
```
Comment (joel-hamill) [inline swagger.yml]: Add period

File: swagger.yml [hunk 13]
```diff
@@ -0,0 +1,218 @@
...
+        401:
+          description: Not authorized
```
Comment (joel-hamill) [inline swagger.yml]: Add period

File: swagger.yml [hunk 14]
```diff
@@ -0,0 +1,218 @@
...
+        500:
+          description: Internal server error
```
Comment (joel-hamill) [inline swagger.yml]: Add period

File: swagger.yml [hunk 15]
```diff
@@ -0,0 +1,218 @@
...
+        - $ref: "#/parameters/limit"
+        - $ref: "#/parameters/skip_next"
+        - $ref: "#/parameters/skip_prev"
+        - $ref: "#/parameters/cursor"
+        - $ref: "#/parameters/read_reverse"
+        - $ref: "#/parameters/postfix"
+          
+      responses:
+        200:
+          description: Successful response
```
Comment (joel-hamill) [inline swagger.yml]: Global comment: add period to all descriptions.

**Comment by darkonie** [comment]
@joel-hamill updated

**Comment by joel-hamill** [comment]
lgtm - i sent link to Emily on the docs team for her input as well (for some reason I can't `@` her in the PR).

**Comment by emanic** [comment]
Hi @darkonie This looks really nice, excited to add this YAML to our collection. One thing that we are using in our YAML is `tags`, which provide the optimal display on the doc site. This is also used in the reference Swagger Pet Store example. You can see this treatment in the Marathon and IAM APIs. It is very easy to implement, you just declare the `tags` in the beginning, I think for this API they would be: `range`, `fields`, and `stream`. Then in each method, you identify which tag it belongs to. Please see https://github.com/dcos/dcos-docs/blob/master/1.9/api/marathon.yaml for an example, although this is not the best example because it is missing the `description` of each tag and has spurious `^M` characters. A better example is the IAM YAML, which you can see here https://raw.githubusercontent.com/mesosphere/dcos-docs-enterprise/master/1.9/api/iam.yaml?token=AIq0KO2qNYggE5zqTeRZDRP2f8gjFyU5ks5YvwlBwA%3D%3D. Please let me know if you have any questions. In addition, I would strongly recommend changing the name of the API to be just "Logging API". Please let me know if you have any objection. One more thing that I would say is that we are not really using descriptions of the kind that you have at the top. Actually, we include this information in the surrounding text and not in the YAML itself. This content is really great, it's just something we would include elsewhere, none of our other APIs have this. Thanks so much for your work on this! It really looks great. 👍 💯

**Comment by darkonie** [comment]
@emanic i was following example from [dcos-metrics](https://github.com/dcos/dcos-metrics/blob/master/docs/HTTP_API.yml). Can we merge this as is and then open PRs to make some clean ups and optimize the yaml structure?

File: swagger.yml [hunk 16]
```diff
@@ -0,0 +1,217 @@
+swagger: '2.0'
+
+info:
+  version: "0.0.1"
+  title: DC/OS Logging Service HTTP API
```
Comment (karlkfi) [inline swagger.yml]: title: DC/OS Log API

File: swagger.yml [hunk 17]
```diff
@@ -0,0 +1,217 @@
...
+  description: >
+    The first version of the DC/OS Logging Service HTTP API.
```
Comment (karlkfi) [inline swagger.yml]: `DC/OS Log API`

File: swagger.yml (hunk 16 above)
Comment (darkonie) [inline swagger.yml]: done

File: swagger.yml (hunk 17 above)
Comment (darkonie) [inline swagger.yml]: done

**Comment by joel-hamill** [comment]
lgtm 🚀

**Pull request merged**